solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1018,8 +1040,52 @@ private void loadInternal() {
         "unloaded",
         SolrInfoBean.Category.CONTAINER.toString(),
         "cores");
+
     Path dataHome =
         cfg.getSolrDataHome() != null ? cfg.getSolrDataHome() : cfg.getCoreRootDirectory();
+
+    solrMetricsContext.observableLongGauge(
+        "solr_cores_filesystem_disk_space",
+        "Disk metrics for Solr's data home directory",
+        measurement -> {
+          try {
+            measurement.record(
+                Files.getFileStore(dataHome).getTotalSpace(),
+                containerAttrs.toBuilder().put(TYPE_ATTR, "total_space").build());
+            measurement.record(
+                Files.getFileStore(dataHome).getUsableSpace(),
+                containerAttrs.toBuilder().put(TYPE_ATTR, "usable_space").build());
+          } catch (IOException e) {
+            throw new SolrException(
+                ErrorCode.SERVER_ERROR,
+                "Error retrieving disk space information for data home directory" + dataHome,
+                e);
+          }
+        },
+        OtelUnit.BYTES);
+    solrMetricsContext.observableLongGauge(
+        "solr_cores_root_disk_space",
```

Comment:
I suggest dropping this as redundant with "data".  If someone defined "data" (which is maybe rare and perhaps should be unsupported), then this different directory would be insignificant, only holding metadata.